### PR TITLE
added logging callback support

### PR DESCRIPTION
**Pull request by snudler6**

### CODE REVIEW

File: src/tracy.c [hunk 1]
```diff
@@ -101,6 +105,11 @@ void TRC_log_and_clear_on_error(TRC_err err) {
 }
 
 
+void TRC_register_err_print_callback(TRC_err_print_callback callback) {
+	print_callback = callback;
```
Comment (FarmerSez) [inline src/tracy.c]: Over-indented.

File: src/tracy.c [hunk 2]
```diff
@@ -152,3 +161,15 @@ static char const * get_error_string(int err) {
   return strerror_r(err, errbuf, MAX_ERR_STR_BUF_SIZE);
 #endif
 }
+
+void trc_private_print(char const * fmt, ...) {
+  va_list args;
+  va_start(args, fmt);
+  if (print_callback) {
```
Comment (FarmerSez) [inline src/tracy.c]: No need for special cases. Write a default print callback that uses `vprintf` and set it to the print callback. Then you should just assert that the callback handler is not NULL.

File: src/tracy.c (hunk 1 above)
Comment (FarmerSez) [inline src/tracy.c]: Check that it's not NULL.

File: src/main.c [hunk 3]
```diff
@@ -1,24 +1,49 @@
...
 TRC_err func3(void) {
   TRC_RETURN_ON_ERROR(func2());
   printf("should not be printed\n");
   return TRC_OK;
 }
 
+
 TRC_err main(void) {
+
+  TRC_register_err_print_callback(print_err_in_red);
```
Comment (FarmerSez) [inline src/main.c]: Add documentation to the readme.

File: src/tracy.c (hunk 1 above)
Comment (snudler6) [inline src/tracy.c]: Done.

Comment (snudler6) [inline src/tracy.c]: Done.

File: src/tracy.c (hunk 2 above)
Comment (snudler6) [inline src/tracy.c]: Done.

**Comment by snudler6** [comment]
Can see the readme changes compiled here: https://github.com/snudler6/tracy

File: src/main.c (hunk 3 above)
Comment (snudler6) [inline src/main.c]: Done.

File: README.md [hunk 4]
```diff
@@ -149,4 +149,16 @@ TRC_CLEANUP_RETURN_ON_ERROR(err, cleanup)
...
+registering of different output handlers using:
+
+```c
+typedef void (*TRC_err_print_callback)(char const * fmt, va_list vargs);
+
+/* Register an output handler for Tracy's outputs */
+void TRC_register_err_print_callback(TRC_err_print_callback callback);
+```
+
+Sending NULL to this function will re-register the default tracy
```
Comment (FarmerSez) [inline README.md]: Sentence is incomplete.

**Pull request merged**